### PR TITLE
Namespace the root_path variable

### LINKED ISSUE #68: node.run_state['root_path'] conflicts with rbenv cookbook
## :ghost: Brief Description

This is a companion issue to this rbenv issue: sous-chefs/ruby_rbenv#285

When using both the rbenv and pyenv cookbooks together and installing python and ruby versions for the same user, the resources from the 2nd cookbook do not work. This is because the same internal run_state attribute is used for tracking the root_path in both cookbooks.

## :pancakes: Cookbook version

3.2.0

## :woman_cook:  Chef-Infra Version

14.13.11

## :tophat: Platform details

RHEL7

## Steps To Reproduce

1. Write cookbook that creates a user and installs a ruby and python version for that user:
```ruby
     user 'jenkins' do
       manage_home true
     end

     rbenv_user_install 'jenkins'

     rbenv_ruby '2.7.1' do
       user 'jenkins'
     end

     pyenv_user_install 'jenkins'

     pyenv_python '3.8.2' do
       user 'jenkins'
     end
```
2. Run the recipe against a node in test-kitchen or a test env.
3. See the pyenv python installation fail

## :police_car: Expected behavior

I expect that both the ruby and python versions are installed with no error.

## :heavy_plus_sign: Additional context

I can work around the problem with a ruby block in between the pyenv and rbenv resources:

```ruby
ruby_block 'unset run_state' do
  block do
    node.run_state['root_path'] = {}
  end
end
```

### PR DESCRIPTION
**Pull request by petracvv**
Signed-off-by: Mihai Petracovici <petracvv@users.noreply.github.com>

# Description

Namespaces the root_path variable so that it does not conflict with other cookbook run_state variables.

## Issues Resolved

#68 

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: libraries/chef_pyenv_script_helpers.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@ module Pyenv
     module ScriptHelpers
       include Chef::Mixin::ShellOut
       def root_path
-        node.run_state['root_path'] ||= {}
+        node.run_state['sous-chefs']['pyenv']['root_path'] ||= {}
```
Comment (Xorima) [inline libraries/chef_pyenv_script_helpers.rb:8]: I am pretty sure you have to initialise each level in `run_state` so you will need to do this:

```
         node.run_state['sous-chefs'] ||= {} 
         node.run_state['sous-chefs']['pyenv'] ||= {} 
         node.run_state['sous-chefs']['pyenv']['root_path'] ||= {} 
```

Comment (petracvv) [reply libraries/chef_pyenv_script_helpers.rb:8]: Yeah I just realized that

**Comment by petracvv** [comment]
Looks like the tests are running into this bug now which was fixed in the ruby_rbenv cookbook:  sous-chefs/ruby_rbenv#282

**Comment by ramereth** [comment]
@petracvv I just reran the jobs and they pass now. I believe an upstream bug with chef was finally resolved that wear causing these issues.

File: libraries/chef_pyenv_script_helpers.rb [hunk 2]
```diff
@@ -5,12 +5,14 @@ module Pyenv
     module ScriptHelpers
       include Chef::Mixin::ShellOut
       def root_path
-        node.run_state['root_path'] ||= {}
+        node.run_state['sous-chefs'] ||= {}
+        node.run_state['sous-chefs']['pyenv'] ||= {}
+        node.run_state['sous-chefs']['pyenv']['root_path'] ||= {}
```
Comment (ramereth) [inline libraries/chef_pyenv_script_helpers.rb:10]: Actually, shouldn't this be in the `pyenv` namespace since we own that namespace on supermarket?

Comment (Xorima) [reply libraries/chef_pyenv_script_helpers.rb:10]: This is actually better imo. We might own the `pyenv` but by using `sous-chefs` we reduce the risk of clashing to basically 0 while using the cache.

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by Xorima** [comment]
@petracvv Can you get a Changelog entry for this and we can get it merged and shipped today 

Thanks!

**Comment by petracvv** [comment]
OK. I added one with a minor version bump in the changelog file. Feel free to edit that if that is not the version you were aiming for

**Pull request merged**